src/main/java/com/linecorp/armeria/common/thrift/text/TTextProtocol.java

```diff
@@ -558,7 +562,23 @@ public short readI16() throws TException {
 
     @Override
     public int readI32() throws TException {
-        return readNameOrValue(TypedParser.INTEGER);
+        if (currentFieldEnumClass != null) {
+            // Enum fields may be set by string, even though they represent integers.
+            getCurrentContext().read();
+            JsonNode elem = getCurrentContext().getCurrentChild();
+            if (elem.isInt()) {
+                return TypedParser.INTEGER.readFromJsonElement(elem);
+            } else if (elem.isTextual()){
+                @SuppressWarnings("rawtypes,unchecked") // All TEnum are enums
+                Class casted = (Class) currentFieldEnumClass;
+                TEnum tEnum = (TEnum) Enum.valueOf(casted, TypedParser.STRING.readFromJsonElement(elem));
+                return tEnum.getValue();
+            } else {
+                throw new TTransportException("Invalid value type for enum field.");
```

Comment:
- `Invalid` -> `invalid`
- `"invalid value type for enum field: " + elem.getNodeType() + " (" + elem + ')'`